aws/rust-runtime/aws-config/src/ecs.rs

```diff
@@ -93,15 +104,32 @@ impl EcsCredentialsProvider {
 
     /// Load credentials from this credentials provider
     pub async fn credentials(&self) -> provider::Result {
-        let auth = match self.env.get(ENV_AUTHORIZATION).ok() {
-            Some(auth) => Some(HeaderValue::from_str(&auth).map_err(|err| {
-                tracing::warn!(token = %auth, "invalid auth token");
+        let env_token = self.env.get(ENV_AUTHORIZATION_TOKEN).ok();
+        let env_token_file = self.env.get(ENV_AUTHORIZATION_TOKEN_FILE).ok();
+        let auth = if let Some(auth_token) = env_token {
```

Comment:
`AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE` is supposed to take priority over `AWS_CONTAINER_AUTHORIZATION_TOKEN`, so these conditions will need to get swapped, and the doc comment updated. There should be a test that sets both to exercise the precedence too.